addons/http_routing/models/ir_http.py

```diff
@@ -335,6 +158,179 @@ def _get_converters(cls):
             model=ModelConverter,
         )
 
+    # ------------------------------------------------------------
+    # Language tools
+    # ------------------------------------------------------------
+
+    @classmethod
+    def _url_localized(cls,
+            url: typing.Optional[str] = None,
+            lang_code: typing.Optional[str] = None,
+            canonical_domain: typing.Optional[typing.Union[str, typing.Tuple[str, str, str, str, str]]] = None,
```

Comment:
Python 3.10 enables to write
```suggestion
            url: str | None = None,
            lang_code: str | None = None,
            canonical_domain: str | tuple[str, str, str, str, str] | None = None,
```